authentication/provider/service/authentication_provider_service.go

```diff
@@ -363,6 +363,12 @@ func (s *authenticationProviderServiceImpl) UpdateIdentityUsingUserInfoEndPoint(
 		return nil, errors.New("unable to get user profile " + err.Error())
 	}
 
+	// Set the user profile's username in the context
+	profileCtx := ctx.Value(provider.UserProfileContextKey)
+	if profileCtx != nil {
+		profileCtx.(*provider.UserProfileContext).Username = &userProfile.Username
```

Comment:
How about using
```go
if pCtx, ok := profileCtx.(*provider.UserProfileContext); ok {
pCtx.Username = &userProfile.Username
``` 
to avoid panic trigger?